phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -304,15 +305,21 @@ public static ConnectionInfo create(String url) throws SQLException {
                     }
                 }
             }
-            return new ConnectionInfo(quorum,port,rootNode, principal, keytabFile);
+            return new ConnectionInfo(quorum,port,rootNode, principal, keytabFile, tenantID);
         }
-        
+
         public ConnectionInfo normalize(ReadOnlyProps props, Properties info) throws SQLException {
+            return normalize(props, info, null);
+        }
+
+        public ConnectionInfo normalize(ReadOnlyProps props, Properties info, String url)
+                throws SQLException {
             String zookeeperQuorum = this.getZookeeperQuorum();
             Integer port = this.getPort();
             String rootNode = this.getRootNode();
             String keytab = this.getKeytab();
             String principal = this.getPrincipal();
+            String tenantID = getTenantID(url);
```

Comment:
nit: This variable seems redundant can just pass to the ConnectionInfo constructor.